libraries/tools/kotlin-gradle-plugin-integration-tests/src/test/kotlin/org/jetbrains/kotlin/gradle/KaptIncrementalWithIsolatingApt.kt

```diff
@@ -403,6 +403,23 @@ class KaptIncrementalWithIsolatingApt : KaptIncrementalIT() {
             }
         }
     }
+
+    @DisplayName("KT-41456: detect missing kotlin compilation output")
+    @GradleTest
+    @DisabledOnOs(OS.WINDOWS, disabledReason = "https://youtrack.jetbrains.com/issue/KTI-405")
+    fun testMissingKotlinOutputForcesNonIncrementalRun(gradleVersion: GradleVersion) {
+        kaptProject(gradleVersion) {
+            build("clean", "assemble")
+
+            // Explicitly remove all kotlinc generated .class files
+            projectPath.resolve("build/classes/kotlin/main").toFile().listFiles()!!.forEach { it.deleteRecursively() }
```

Comment:
Nit: there is [a helper method](https://github.com/JetBrains/kotlin/blob/9943b146e985195df1b6f18cbc03851d0bd3c6b1/libraries/tools/kotlin-gradle-plugin-integration-tests/src/test/kotlin/org/jetbrains/kotlin/gradle/testbase/pathHelpers.kt#L89) for `Path` to delete recusively

Reply:
I don't think that keeps build/classes/kotlin/main, and I wanted to keep the top-level dir (to simulate some classes being removed but top-level dir still being there).